python_modules/dagster/dagster/core/definitions/schedule_definition.py

```diff
@@ -433,22 +476,29 @@ def evaluate_tick(self, context: "ScheduleEvaluationContext") -> ScheduleExecuti
         from .decorators.schedule_decorator import DecoratedScheduleFunction
 
         check.inst_param(context, "context", ScheduleEvaluationContext)
+        execution_fn: Callable[[ScheduleEvaluationContext], "ScheduleEvaluationFunctionReturn"]
         if isinstance(self._execution_fn, DecoratedScheduleFunction):
             execution_fn = self._execution_fn.wrapped_fn
         else:
-            execution_fn = cast(Callable[[ScheduleEvaluationContext], Any], self._execution_fn)
+            execution_fn = cast(
+                Callable[[ScheduleExecutionContext], "ScheduleEvaluationFunctionReturn"],
+                self._execution_fn,
+            )
+
         result = list(ensure_gen(execution_fn(context)))
 
         skip_message: Optional[str] = None
 
+        run_requests: List[RunRequest] = []
         if not result or result == [None]:
             run_requests = []
             skip_message = "Schedule function returned an empty result"
         elif len(result) == 1:
-            item = result[0]
-            check.inst(item, (SkipReason, RunRequest))
-            run_requests = [item] if isinstance(item, RunRequest) else []
-            skip_message = item.skip_message if isinstance(item, SkipReason) else None
+            item = check.inst(result[0], (SkipReason, RunRequest))
+            if isinstance(item, RunRequest):
+                run_requests, skip_message = [item], None
+            elif isinstance(item, SkipReason):
+                run_requests, skip_message = [], item.skip_message
```

Comment:
nit: hmm not sure how i feel about compact assign like this vs just doing multi line. feel like its harder to parse at a glance. Maybe just a preference thing.

Reply:
The reason I did this was that for some reason type narrowing (in mypy) was not working as expected with the previous form, leading to a type error.